great_tables/_formats.py

```diff
@@ -1771,6 +1771,209 @@ def fmt_time_fn(
+        **Great Tables** will choose `True` when `as_button=False` and `False` in the other case.
+        The link underline will be the same color as that set in the `color` option.
+    button_fill : str
+        The color to fill the button with. This is ignored if `as_button` is `False`. The default is
+        `"auto"` which will use the default background color for the button.
+    button_width : str
+        The width of the button. This is ignored if `as_button` is `False`. The default is `"auto"`
+        which will use the default button width for link-as-button.
+    button_outline : str
+        Options for styling a link-as-button (and only applies if `as_button=True`). This option is
+        by default set to `"auto"`, allowing **Great Tables** to choose the appropriate outline
+        color value.
+
+    Returns
+    -------
+    GT
+        The GT object is returned.
+    """
+
+    if as_button:
+        #
+        # All determinations of `color`, `show_underline`, `button_fill` and
+        # `button_width` for the case where `as_button=True`; each of the
+        # above arguments are set to "auto" by default
+        #
+
+        # In the button case, we opt to never show an underline unless it's
+        # requested by the user (i.e., `show_underline=True`)
+        if show_underline == "auto":
+            show_underline = False
+
+        if button_width == "auto":
+            button_width = None
+
+        button_outline_color = button_outline
+        button_outline_style = "solid"
+        button_outline_width = "2px"
+
+        # There are various combinations of "auto" or not with `button_fill` and
+        # `color` that need to be handled delicately so as to ensure contrast
+        # between foreground text and background fill is maximized
+        if button_fill == "auto" and color == "auto":
+            # Choose a fixed and standard color combination if both options are
+            # 'auto'; these will be 'steelblue' and 'white'
+            button_fill = "#4682B4"
+            color = "#FFFFFF"
+
+        elif button_fill == "auto" and color != "auto":
+            # If `button_fill` is 'auto' but `color` is not, then we need to
+            # determine whether the background should be light or dark
+            bgrnd_bw = "#FFFFFF"
+
+            if bgrnd_bw == "#FFFFFF":
+                # Background should be light so using 'lightblue'
+                button_fill = "#ADD8E6"
+            else:
+                # Background should be dark so using 'darkblue'
+                button_fill = "#00008B"
+
+            if button_outline == "auto":
+                button_outline_color = "#BEBEBE"
+                button_outline_style = "none"
+
+        elif button_fill != "auto" and color == "auto":
+            if button_outline == "auto":
+                button_outline_color = "#DFDFDF"
+
+                if button_fill in [
```

Comment:
I wonder if it'd be helpful to define lists like this at the top of the function, with an informative name like...

LIGHT_FILL_VALUES